.github/workflows/ci.yml

```diff
@@ -0,0 +1,93 @@
+name: Alfresco Java SDK CI
+
+on:
+  push:
+    branches:
+      - release/5.1.N
+  pull_request:
+    branches:
+      - release/5.1.N
+  workflow_dispatch:
+
+env:
+  MAVEN_PASSWORD: ${{ secrets.NEXUS_PASSWORD }}
+  MAVEN_USERNAME: ${{ secrets.NEXUS_USERNAME }}
+  GITHUB_ACTIONS_DEPLOY_TIMEOUT: 60
+  CONTENT_SERVICE_SECURITY_BASICAUTH_PASSWORD: ${{ secrets.CONTENT_SERVICE_SECURITY_BASICAUTH_PASSWORD }}
+  CONTENT_SERVICE_SECURITY_BASICAUTH_USERNAME: ${{ secrets.CONTENT_SERVICE_SECURITY_BASICAUTH_USERNAME }}
+  CONTENT_SERVICE_URL: ${{ secrets.CONTENT_SERVICE_URL }}
+  ENVIRONMENT_APA_HOST: apadev-apa.envalfresco.com
+  ENVIRONMENT_HOST: apadev.envalfresco.com
+  ENVIRONMENT_APP: simpleapp
+  AGS_VERSION: 14.9
+  ACS_VERSION: 7.1.0.1
+  JAVA_VERSION: 11
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: Alfresco/alfresco-build-tools/.github/actions/pre-commit@v1.35.0
+
+  build_and_verify:
+    name: "Build and Verify"
+    runs-on: ubuntu-latest
+    needs:
+      - pre-commit
+    env:
+      MAVEN_CLI_OPTS: >
+        source:jar -B -q -e -fae -V -DinstallAtEnd=true -U -Ddoclint=none
+        -Denvironment.host=${ENVIRONMENT_HOST}
+        -Denvironment.apa.host=${ENVIRONMENT_APA_HOST}
+        -Denvironment.application.name=${ENVIRONMENT_APP}
+        -Dags.version=${AGS_VERSION}
+        -Dacs.version=${ACS_VERSION}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: Alfresco/alfresco-build-tools/.github/actions/get-build-info@v1.35.0
+      - uses: Alfresco/alfresco-build-tools/.github/actions/setup-java-build@v1.35.0
+        with:
+          java-version: ${{ env.JAVA_VERSION }}
+      - name: "Generate ACS Rest API"
+        timeout-minutes: ${{ fromJSON(env.GITHUB_ACTIONS_DEPLOY_TIMEOUT) }}
+        run: |
+          cd alfresco-acs-java-rest-api/alfresco-acs-java-rest-api-lib
+          bash generate.sh
+      - name: "Generate APA Rest API"
+        timeout-minutes: ${{ fromJSON(env.GITHUB_ACTIONS_DEPLOY_TIMEOUT) }}
+        run: |
+          cd alfresco-apa-java-rest-api/alfresco-apa-java-rest-api-lib
+          bash generate.sh
```

Comment:
These steps are also needed for the publish job as far as I understood from the original Travis implementation which re-uses the workspace with these files already generated there. cc: @jesty, correct me if I'm wrong.